Modules/Filtering/BiasCorrection/include/itkN4BiasFieldCorrectionImageFilter.hxx

```diff
@@ -66,11 +67,49 @@ N4BiasFieldCorrectionImageFilter<TInputImage, TMaskImage, TOutputImage>
   this->m_MaximumNumberOfIterations.Fill( 50 );
 }
 
+
+template<typename TInputImage, typename TMaskImage, typename TOutputImage>
+void
+N4BiasFieldCorrectionImageFilter<TInputImage, TMaskImage, TOutputImage>
+::VerifyPreconditions() ITKv5_CONST
+{
+  Superclass::VerifyPreconditions();
+
+  const InputImageType *const inputImage = this->GetInput();
+
+  if (inputImage == nullptr)
+  {
+    itkExceptionMacro("Failed precondition: Input image should not be null");
+  }
+
+  const typename InputImageType::SizeType inputImageSize = inputImage->GetBufferedRegion().GetSize();
+
+  const MaskImageType *const maskImage = GetMaskImage();
+
+  if ((maskImage != nullptr) && (maskImage->GetBufferedRegion().GetSize() != inputImageSize))
+  {
+    itkExceptionMacro(
+      "Failed precondition: If a mask image is specified, its size should be equal to the input image size");
+  }
+
+  const RealImageType *const confidenceImage = GetConfidenceImage();
+
+  if ((confidenceImage != nullptr) && (confidenceImage->GetBufferedRegion().GetSize() != inputImageSize))
+  {
+    itkExceptionMacro(
+      "Failed precondition: If a confidence image is specified, its size should be equal to the input image size");
```

Comment:
Similar here. Please review [section 8.3.1 of the ITK Software guide](https://itk.org/ITKSoftwareGuide/html/Book1/ITKSoftwareGuide-Book1ch8.html#x54-1330008) explaining the step and how the ITK pipeline works by default. By default this filter's request output region is propagated to all the inputs in the substeps of `PropagateRequestedRegion()`. I'm not sure what the issue was to add these checks, but knowing that we can then work on a proper solution with the ITK pipeline.